### PR TITLE
EKF control minor cleanup (use switch statement and fix code style)

### PR DESCRIPTION
**Pull request by dagar**
I was going to take a look at aiding range aid for vision height mode, but thought I'd get this out of the way first.

### CODE REVIEW

File: EKF/control.cpp [hunk 1]
```diff
@@ -931,8 +931,11 @@ void Ekf::controlHeightFusion()
 	checkRangeAidSuitability();
 	_range_aid_mode_selected = (_params.range_aid == 1) && isRangeAidSuitable();
 
-	if (_params.vdist_sensor_type == VDIST_SENSOR_BARO) {
-
+	switch (_params.vdist_sensor_type) {
+	default:
```
Comment (kamilritz) [inline EKF/control.cpp]: What is the default case added for? Why the fallthrough?

File: EKF/control.cpp [hunk 2]
```diff
@@ -976,10 +979,9 @@ void Ekf::controlHeightFusion()
 				_hgt_sensor_offset = _gps_sample_delayed.hgt - _gps_alt_ref + _state.pos(2);
 			}
 		}
-	}
+		break;
 
-	// set the height data source to range if requested
-	if ((_params.vdist_sensor_type == VDIST_SENSOR_RANGE) && _rng_hgt_valid) {
```
Comment (kamilritz) [inline EKF/control.cpp]: The check for the `_rng_hgt_valid` is missing now

File: EKF/control.cpp (hunk 1 above)
Comment (dagar) [reply EKF/control.cpp]: I thought it couldn't hurt if someone entered a bogus HGT_MODE.

File: EKF/control.cpp (hunk 2 above)
Comment (dagar) [reply EKF/control.cpp]: Oops, nice catch. Fixed.

I also changed it to _range_data_ready && _rng_hgt_valid for consistency.

**Pull request merged**